### DISCUSSION TITLE
Feature: Add the ability to query for all global plugins

### DISCUSSION THREAD

**Discussion by ingshtrom** [Kong/kong#7249] [Ideas and feature requests]
### Summary

I have the need to query for all plugins where api_id=(nothing) but I can't find a way to do it now. It seems that either the crud_helpers or daos are validating the input as uuid and not allowing other values. While I understand this validation, it causes problems for our provisioning/scheduling application because now we need to get EVERY plugin (and it's config), send that over the wire, and then filter it down in memory... I can only imagine how painful this will become once we start getting enough plugins that we have to page that API... ><

I could probably do it myself, but I need the OK from the team that this is a feature you guys want to support.

OOOORRRRRR, maybe I just missed this and it's already a feature!

Thanks for all of your hard work. Kong is working wonders for our ability to route requests and be able to do AB Testing.

**Comment by thibaultcha** [comment]
I think this is indeed missing. Contributions welcome!

**Comment by subnetmarco** [comment]
Isn't `/plugins` a way to do that?

**Comment by thibaultcha** [comment]
I believe `/plugins` will list **all** plugins, while @ingshtrom wants to retrieve all _global_ plugins (ones that apply to all of the APIs)?

**Comment by ingshtrom** [comment]
@thibaultcha exactly

@thefosk Could you show me an example? I'm on v0.9.0 and /plugins shows me all plugins, including the ones that are attached to APIs. I only want to see _global_ plugins.

Example:

I have 100 plugins. 95 are attached to APIs and 5 are global. I hit `/plugins` and I get a LOT of JSON back when I only need 5% of the returned JSON (the 5 global plugins). I would think it would be more efficient to just write the db query so that we only return the bare necessity.

Don't get me wrong, I am using `/plugins` right now and it works perfectly fine, but I worry about the efficiency of it down the road.

**Comment by Tieske** [comment]
@ingshtrom off topic; just curious, what's your setup for AB testing with Kong?

**Comment by ingshtrom** [comment]
@Tieske basically we have a header that we rewrite into the url (or set a default value) and then Kong routes via `request_path`. This isn't in production, but we've had success with simple POCs--in fact our Kong implementation isn't in prod yet either, but getting close!

**Comment by hbagdi** [comment]
I took a stab at this but was not successful.

In our `plugins` table, records which have `api_id` and `consumer_id` as null are global plugins.

I could query for all plugin and then filter out which is a very inefficient way to do things.

I could not query (using the crud helpers) from the DB itself. Like:
```sql
SELECT *
FROM plugins
WHERE api_id IS NULL AND
consumer_id IS NULL;
```

Is this even possible?
